Adds a few deprecated features and one new feature to authentication … (#41535)
openshift/openshift-docs: release_notes/ocp-4-10-release-notes.adoc

[id="ocp-4-10-auth"]
=== Authentication and authorization

[id="ocp-4-10-oc-commands-obtain-credentials-from-podman-config"]

> ==== `oc` commands now obtain credentials from podman configuration locations
> 
> Previously, `oc` commands that use the registry configuration, for example `oc login` or `oc image`, obtained credentials from Docker configuration locations. With {product-title} {product-version}, if a registry entry cannot be found in the default Docker configuration location, `oc` commands obtain the credentials from podman configuration locations. You can set your preference to either Docker or podman by using the `REGISTRY_AUTH_PREFERENCE` environment variable or the `--registry-config` flag.

Reviewer: We generally capitalize Podman as a noun and use lowercase to describe the `podman` command.

```suggestion
==== `oc` commands now obtain credentials from Podman configuration locations

Previously, `oc` commands that use the registry configuration, for example `oc login` or `oc image`, obtained credentials from Docker configuration locations. With {product-title} {product-version}, if a registry entry cannot be found in the default Docker configuration location, `oc` commands obtain the credentials from podman configuration locations. You can set your preference to either Docker or Podman by using the `REGISTRY_AUTH_PREFERENCE` environment variable or the `--registry-config` flag.
```

> 

Reviewer 2: > or the `--registry-config` flag.

- this points to the single config file path, not the preference. In contrast each preference goes through multiple files.
- this feature already exists in oc

Reviewer 2: > You can set your preference to either Docker or podman by using the 

Can we rewrite it like this to highlight the values?

You can set your preference to either `docker` or `podman` by using the

Reviewer 2: ah please note the difference between `REGISTRY_AUTH_FILE` and `REGISTRY_AUTH_PREFERENCE`. `REGISTRY_AUTH_FILE` is indeed similar to `--registry-config`

[id="ocp-4-10-auth"]
=== Authentication and authorization

[id="ocp-4-10-oc-commands-obtain-credentials-from-podman-config"]
==== oc commands now obtain credentials from Podman configuration locations

Previously, `oc` commands that used the registry configuration, for example `oc login` or `oc image`, obtained credentials from Docker configuration locations. With {product-title} {product-version}, if a registry entry cannot be found in the default Docker configuration location, `oc` commands obtain the credentials from Podman configuration locations. You can set your preference to either `docker` or `podman` by using the `REGISTRY_AUTH_PREFERENCE` environment variable to prioritize the location.

> Users also have the option to use the `REGISTRY_AUTH_FILE` parameter, which serves as an alternative to the existing `--registry-config` CLI flag. `REGISTRY_AUTH_FILE` is also compatible with `podman` variables.

Reviewer 2: ```suggestion
Users also have the option to use the `REGISTRY_AUTH_FILE` env variable, which serves as an alternative to the existing `--registry-config` CLI flag. `REGISTRY_AUTH_FILE` env variable is also compatible with `podman`.
```

or env -> environment

AMD and Intel 64-bit architectures (x86-64-v2) will still be supported.
====

[id="ocp-4-10-docker-config-location-deprecation"]

> ==== Default Docker configuration location Deprecated

Reviewer: Sentence case:
```suggestion
==== Default Docker configuration location deprecated
```

Or you could write "deprecation" if you wanted to match the next two entries' style 🤷

> Previously, `oc` commands that used a registry configuration obtained credentials from the Docker configuration location, which was `~/.docker/config.json` by default. This has been deprecated and will be replaced by a podman configuration location in a future version of {product-title}.

Reviewer: I first tried to read this as "registry configuration-obtained", so a possible rewording for clarity could be:
```suggestion
Previously, `oc` commands that used a registry configuration would obtain credentials from the Docker configuration location, which was `~/.docker/config.json` by default. This has been deprecated and will be replaced by a Podman configuration location in a future version of {product-title}.
```

[id="ocp-4-10-empty-file-support-deprecation"]
==== Empty file support deprecation

Support for empty files using the `--registry-config` and `--to` flags in `oc registry login` has been deprecated. It will be removed in a future version of {product-title}.

[id="ocp-4-10-writing-to-stdout-option-deprecation"]
==== Return support for `-` and `stdout` deprecation

> Return support for `-` and `stdout` in the `oc registry login` command been deprecated. It will be removed in a future version of {product-title}.

Reviewer 2: Maybe we should mention the context? 

- Which is the same as above:  the `--registry-config` and `--to` flags 
- does it make sense to squash these two sections together?

AMD and Intel 64-bit architectures (x86-64-v2) will still be supported.
====

[id="ocp-4-10-docker-config-location-deprecation"]
==== Default Docker configuration location deprecation
Previously, `oc` commands that used a registry configuration would obtain credentials from the Docker configuration location, which was `~/.docker/config.json` by default. This has been deprecated and will be replaced by a Podman configuration location in a future version of {product-title}.

[id="ocp-4-10-empty-file-support-deprecation"]
==== Empty file and stdout support deprecation in oc registry login

> Support for empty files using the `--registry-config` and `--to` flags in `oc registry login` has been deprecated. Support for `-` (or `stdout`) has also been deprecated as an argument when using `oc registry login`. They will be removed in a future version of {product-title}.

Reviewer 2: ```suggestion
Support for empty files using the `--registry-config` and `--to` flags in `oc registry login` has been deprecated. Support for `-` (stdout) has also been deprecated as an argument when using `oc registry login`. They will be removed in a future version of {product-title}.
```

probably better. stdout is just a shortcut for standard output. I suppose you can use that instead

AMD and Intel 64-bit architectures (x86-64-v2) will still be supported.
====

[id="ocp-4-10-docker-config-location-deprecation"]
==== Default Docker configuration location deprecation
Previously, `oc` commands that used a registry configuration would obtain credentials from the Docker configuration location, which was `~/.docker/config.json` by default. This has been deprecated and will be replaced by a Podman configuration location in a future version of {product-title}.

[id="ocp-4-10-empty-file-support-deprecation"]

> ==== Empty file support deprecation

Reviewer 2: ```suggestion
==== Empty file & stdout support deprecation in oc registry login
```

maybe?

> Support for empty files using the `--registry-config` and `--to` flags in `oc registry login` has been deprecated. Return support for `-` and `stdout` has also been deprecated in `oc registry login`. They will be removed in a future version of {product-title}.

Reviewer 2: not sure if it is clear what return support means. Deprecated is passing `-` as an argument to `--registry-config` or `--to`.  Stdout is just a description what `-` means.
